honeybee_vtk/config.py

```diff
@@ -5,200 +5,223 @@
+        'Defaults to VTKs default settings.'
     )
 
-    label_fonts: FontConfig = Field(
-        FontConfig(),
-        description='Font parameters for the fonts to be used for the labels on the'
-        ' legend.'
+    preceding_labels: bool = Field(
+        False,
+        description='Boolean value to decide whether the legend title and the'
+        ' legend labels will precede the legend or not.'
     )
 
-    title_fonts: FontConfig = Field(
-        FontConfig(bold=True),
-        description='Font parameters for the fonts to be used in the title of the'
-        ' legend.'
+    label_parameters: TextConfig = Field(
+        TextConfig(),
+        description='Text parameters for the labels on the legend.'
     )
 
-    @validator('colors')
-    def validate_colors(cls, v: str) -> str:
-        colors = dir(Colors)[4:]
-        if v.lower() in colors:
-            return v
-        else:
-            raise ValueError(
-                f'Colors must be a string from {tuple(colors)}. Instead got'
-                f' {v}.'
-            )
-
-    @validator('orientation')
-    def validate_orientation(cls, v: str) -> str:
-        orientations = ('horizontal', 'vertical')
-        if v.lower() in orientations:
-            return v
-        else:
-            raise ValueError(
-                f'Orientation must be from {orientations}. Instead got {v}.'
-            )
-
-    @validator('label_format')
-    def validate_label_format(cls, v: str) -> str:
-        label_formats = dir(LabelFormat)[4:]
-        if v in label_formats:
-            return v
-        else:
-            raise ValueError(
-                f'Label format must be from {label_formats}. Instead got {v}.'
-            )
+    title_parameters: TextConfig = Field(
+        TextConfig(bold=True),
+        description='Text parameters for the title of the legend.'
+    )
 
 
 class DataConfig(BaseModel):
-    """Config for each dataset you'd like to load on a honeybee-vtk model."""
+    """Config for simulation results you'd like to load on a honeybee-vtk model."""
+
+    class Config:
+        validate_all = True
+        validate_assignment = True
 
-    name: str = Field(
-        description='Name to be given to data. Example, "Daylight-Factor".'
+    identifier: str = Field(
+        description='identifier to be given to data. Example, "Daylight-Factor".'
+        ' This identifier needs to be unique in each of the DataConfig objects'
+        ' one introduces using a config file. Having multiple DataConfig objects with'
+        ' same modifier will raise an error.'
     )
 
-    object_type: str = Field(
+    object_type: DataSetNames = Field(
         description='The name of the model object on which you would like to map this'
-        ' data.'
+        ' data. Acceptable values are "wall", "aperture", "shade", "door", "floor",'
+        ' "roofceiling", "airboundary", "grid".'
```

Comment:
Maybe remove this from the description here since you already have the Enumeration to tell you the acceptable values.

Reply:
Complied